sylvia/src/multitest.rs

```diff
@@ -1,62 +1,100 @@
 use std::cell::{Ref, RefCell, RefMut};
+use std::fmt::{Debug, Display};
 use std::marker::PhantomData;
+use std::ops::Deref;
 
-use cosmwasm_std::{Addr, BlockInfo, Coin};
-use cw_multi_test::Executor;
+use cosmwasm_std::{Addr, Api, Coin, CustomQuery, Empty, GovMsg, IbcMsg, IbcQuery, Storage};
+use cw_multi_test::{
+    BankKeeper, DistributionKeeper, Executor, FailingModule, Router, StakeKeeper, WasmKeeper,
+};
+use schemars::JsonSchema;
+use serde::de::DeserializeOwned;
 use serde::Serialize;
 
-#[derive(Default)]
-pub struct App {
-    pub app: RefCell<cw_multi_test::App>,
+pub struct App<MtApp> {
+    app: RefCell<MtApp>,
 }
 
-impl App {
-    pub fn new(app: cw_multi_test::App) -> Self {
-        Self {
-            app: RefCell::new(app),
-        }
+impl<MtApp> Default for App<MtApp>
+where
+    MtApp: Default,
+{
+    fn default() -> Self {
+        Self::new(MtApp::default())
     }
+}
 
-    pub fn app(&self) -> Ref<'_, cw_multi_test::App> {
-        Ref::map(self.app.borrow(), |app| app)
+impl<MtApp> Deref for App<MtApp> {
+    type Target = RefCell<MtApp>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.app
     }
+}
 
-    pub fn app_mut(&self) -> RefMut<'_, cw_multi_test::App> {
-        RefMut::map(self.app.borrow_mut(), |app| app)
+impl<ExecC, QueryC> App<cw_multi_test::BasicApp<ExecC, QueryC>> {
+    /// Creates new default `App` implementation working with customized exec and query messages.
+    pub fn custom<F>(init_fn: F) -> Self
+    where
+        ExecC: Debug + Clone + PartialEq + JsonSchema + DeserializeOwned + 'static,
+        QueryC: Debug + CustomQuery + DeserializeOwned + 'static,
+        F: FnOnce(
+            &mut Router<
+                BankKeeper,
+                FailingModule<ExecC, QueryC, Empty>,
+                WasmKeeper<ExecC, QueryC>,
+                StakeKeeper,
+                DistributionKeeper,
+                FailingModule<IbcMsg, IbcQuery, Empty>,
+                FailingModule<GovMsg, Empty, Empty>,
+            >,
+            &dyn Api,
+            &mut dyn Storage,
+        ),
+    {
+        App {
+            app: RefCell::new(cw_multi_test::custom_app(init_fn)),
+        }
     }
+}
 
-    pub fn block_info(&self) -> BlockInfo {
-        self.app.borrow().block_info()
```

Comment:
I am pretty sure you can just move this `block_info` to the very same impl block as `custom`. You can try if it works, and if so leave it here, if it requires more work just create a new issue from this task for the future.